src/app/modules/shared_models/contract_model.nim

```diff
@@ -0,0 +1,66 @@
+import NimQml, Tables, strutils, stew/shims/strformat
+
+import ./contract_item
+
+type
+  ModelRole {.pure.} = enum
+    Key = UserRole + 1
+    ChainId
+    Address
+
+QtObject:
+  type Model* = ref object of QAbstractListModel
+    items: seq[Item]
+
+  proc setup(self: Model) =
+    self.QAbstractListModel.setup
+
+  proc delete(self: Model) =
+    self.items = @[]
+    self.QAbstractListModel.delete
+
+  proc newModel*(): Model =
+    new(result, delete)
+    result.setup
+    result.items = @[]
+
+  proc `$`*(self: Model): string =
+    for i in 0 ..< self.items.len:
+      result &= fmt"""[{i}]:({$self.items[i]})"""
+
+  method rowCount(self: Model, index: QModelIndex = nil): int =
+    return self.items.len
+
+  proc countChanged(self: Model) {.signal.}
```

Comment:
was decided to not have this on the nim side, will be handled on UI side or rowCount will be used.

Reply:
otoh, if we really need it, it would actually be better to have it here and updated properly

Reply:
hmm true, we should eventually do a batch replacement PR to make sure we use `rowCount` or `ModelCount.count` everywhere to align with the proposal